src/components/search/map/marker.tsx

```diff
@@ -56,8 +58,25 @@ export const Marker = ({
     marker.setFocusHandler((focused: boolean) => onFocus(focused, id));
     marker.setHoverHandler((hovered: boolean) => onHover(hovered, id));
 
+    // Add the pin to the pinStore if it is not a cluster marker
+    const isClusterMarker = id.includes("cluster-");
+    if (cluster && cluster["setPinStore"] && !isClusterMarker) {
+      cluster["setPinStore"]((pinStore) => [
```

Comment:
nit but could we use this syntax instead
```suggestion
      cluster.setPinStore((pinStore) => [
```

makes it easier for me to tell it's a regular function call